File: python/hello_world.py
from mpi4py import MPI

def main():
    # Initialize the MPI environment
    comm = MPI.COMM_WORLD

    # Get the rank of the process and the size of the MPI_COMM_WORLD communicator
    rank = comm.Get_rank()
    size = comm.Get_size()

    # Print a hello world message from each process
    print(f"Hello from process {rank} out of {size}")

if __name__ == "__main__":
    main()

File: cpp/src/hello_world.cpp
#include <mpi.h>
#include <iostream>

int main(int argc, char** argv) {
    // Initialize the MPI environment
    const int INPUT = 100;
    MPI_Init(NULL, NULL);

    // Get the number of processes
    int world_size;
    MPI_Comm_size(MPI_COMM_WORLD, &world_size);

    // Get the rank of the process
    int world_rank;
    MPI_Comm_rank(MPI_COMM_WORLD, &world_rank);

    if(world_rank % 2 == 0) {
        // even
        auto val = INPUT * 2;
        std::cout << "Times 2: " << val << std::endl; 
    } else {
        // odd
        auto val = INPUT * 4;
        std::cout << "Times 4: " << val << std::endl;
    }

    // Print off a hello world message
    

    // Finalize the MPI environment.
    MPI_Finalize();
}
